setup.sh

```diff
@@ -2,6 +2,7 @@
 pushd "$( dirname "${BASH_SOURCE[0]}" )" > /dev/null
 RPI_SETUP_DIR="$( pwd )"
 
+I2S_MODE=
 # Valid values for XMOS device
 XMOS_DEVICE=
 if [[ $# -ge 1 ]]; then
```

Comment:
It looks like this script expects exactly one parameter.  It would be good to add another conditional after this one to detect the case of a use providing two or more parameters.  In all error cases, it is good practice to display the usage line after the error message.